katello/hooks/boot/09-version_locking.rb

```diff
@@ -0,0 +1,8 @@
+app_option(
```

Comment:
One thought to implement this in regular foreman is to check if foreman-maintain is present and only then add the option. Probably best to leave that out of this PR though since it introduces complexities.

Reply:
I made an attempt to handle missing foreman-maintain. Though I guess it is a corner case as this is katello-only and we depend on f-m there.